src/acl/CertificateData.cc

```diff
@@ -107,37 +103,29 @@ ACLCertificateData::parse()
                 return;
             }
 
-            /* an acl must use consistent attributes in all config lines */
-            if (attribute) {
-                if (strcasecmp(newAttribute, attribute) != 0) {
-                    debugs(28, DBG_CRITICAL, "FATAL: An acl must use consistent attributes in all config lines (" << newAttribute << "!=" << attribute << ").");
+            if (attribute.isEmpty() && strcasecmp(newAttribute, "DN") != 0) {
```

Comment:
Can we significantly reduce the diff in this function if we split this `if` statement into two? If yes, let's try that:

```suggestion
            if (attribute.isEmpty()) {
                if (strcasecmp(newAttribute, "DN") != 0) {
```

... and remove minor `auto` improvements (`nid` and `span`) that you have added when the diff was big.